tutorials/themes/tutorials-theme/static/searchtools.js

```diff
@@ -773,86 +820,135 @@ var Search = {
-            {start = firstFound;
-             sentenceStart=0;
-             flag =1;
-             break;}
-            
-          for( sentenceStart = 0; sentenceStart<80; sentenceStart++){ 
-            if((textLower.charAt(start - sentenceStart) == '>') || (textLower.charAt(start - sentenceStart) == '<')) {
-              finderStart = start+1;
-              if(finderStart == startPoint)
-              {firstFound = start;}
-              break;
-            }
-            else if(textLower.charAt(start - sentenceStart) == '.') {
-              flag = 1;
-              break;}
-          }
-          }
-        });
-      start = Math.max(start - sentenceStart, 0);
-      var trimmedText = ((start > 0) ? '...' : '') + $.trim(textLower.substr(start, 350)) + ((start + 350 - textLower.length) ? '...' : '');
+    function cleanupRSTString(rstString) {
+      var newStr;
+
+      // Replace continuous whitespace with single whitespace
+      newStr = rstString.replace(/(\s{2,})/gm, " ");
+      newStr = newStr.replace(/(\r\n|\n|\r)/gm, " ");
+
+      // Replace heading underlines and newlines:
+      newStr = newStr.replace(/(\s[=]{3,}\s)/gm, ": ");
+      newStr = newStr.replace(/(\s[\-]{3,}\s)/gm, ": ");
+
+      // Remove Sphinx directives:
+      newStr = newStr.replace(/(.. [a-zA-Z0-9\-]+::)/gm, "");
+      newStr = newStr.replace(/(:[a-zA-Z0-9\-]+:)/gm, "");
+
+      // Remove URLs:
+      newStr = newStr.replace(/(?:www|https?)[^\s]+/gm, "");
+
+      // Remove RST formatting:
+      newStr = newStr.replace(/(`)/gm, "");
+
+      return newStr;
+    }
+
+    // Start looking at search text below the main header
+    var startIdx = textLower.indexOf('====\n', 0) + 5;
+    var excerpt = "";
+
+    // Content to display on each placard on the tutorials page when a search
+    // query is provided
+    var lines = new Array();
+    if ((keywords == undefined) || (keywords.length == 0)) {
+      // Summary to display on the placards on the tutorials page with no search
+      // query string
+
+      // Find the "Summary" header and text below
+      var summaryHdrIdx = textLower.indexOf("summary", startIdx);
+      var summaryTextIdx = textLower.indexOf("---\n\n", summaryHdrIdx) + 5;
+
+      if (summaryHdrIdx > -1) {  // Summary header found, extract text
+        var parBreakIdx = textLower.indexOf('\n\n', summaryTextIdx);
+        // TODO: this could be customizable - I set it to an arbitrary value...
+        var defaultSummaryLength = 512;
+        var summaryLen = Math.min(defaultSummaryLength,
+                                  parBreakIdx - summaryTextIdx);
+        excerpt = cleanupRSTString(text.substr(summaryTextIdx, summaryLen));
+        excerpt = excerpt + '...';
+
+      } else {  // No summary header found
+        excerpt = 'No tutorial summary found.';
+      }
+    } else {
+      $.each(keywords, function(i, x) {
+        var keyIdx = textLower.indexOf(hlwords[i].toLowerCase(), startIdx);
```

Comment:
When we use the "filters" to search for the results, "hlwords" will look like "filterKEYWORD". Hence, indexof won't find any instance of the hlword.

<img width="1502" alt="Screenshot 2019-09-16 at 2 35 08 PM" src="https://user-images.githubusercontent.com/24625562/64946579-d3b11980-d890-11e9-91ab-631c4178b448.png">

We want to first check if the hlword has "filter" in it. If yes then we have to first strip it, else we can directly use it. Something similar to want we are doing in lines 942 - 950 (i.e. in $.each(hlwords, function())